R/ui_functions.R

```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
   return(out)
 
 }
+
+
+#' Provides numbers that will populate the country profile charts
+#'
+#' @param country character: Country code
+#' @param povline numeric: Poverty line
+#' @param pop_units numeric: Units used to express population numbers (default to million)
+#' @param year_range numeric: Range used to subset countries in Poverty MRV chart
+#' @param lkup list: A list of lkup tables
+#'
+#' @return list
+#' @export
+#'
+ui_cp_charts <- function(country = 'AGO', povline = NULL,
+                         pop_units = 1e6, year_range = 2016:2019,
+                         lkup) {
+
+
+  if (is.null(povline)) {
+    poverty_lines <- lkup$pl_lkup$poverty_line
+    dl <- lapply(poverty_lines, function(pl) {
+      ui_cp_poverty_charts(country = country,
+                           povline = pl,
+                           pop_units = pop_units,
+                           year_range = year_range,
+                           lkup = lkup)
+    })
+    names(dl) <- poverty_lines
+  } else {
+    dl <- ui_cp_poverty_charts(country = country,
+                           povline = povline,
+                           pop_units = pop_units,
+                           year_range = year_range,
+                           lkup = lkup)
+    dl <- list(dl)
+    names(dl) <- povline
+  }
+
+  dl <- list(pov_charts = dl)
+
+  # Fetch pre-calculated data (filter selected country)
+  dl2 <- lapply(lkup$cp$charts, function(x) {
+    x[country_code == country]
+  })
+
+  out <- append(dl, dl2)
+
+  return(out)
+
+}
+
+#' Provides numbers that will populate the country profiles poverty charts
+#'
+#' @inheritParams ui_cp_charts
+#' @return list
+#' @keywords internal
+#'
+ui_cp_poverty_charts <- function(country, povline, pop_units,
+                                 year_range, lkup) {
+
+  # Fetch data for poverty trend chart
+  res_pov_trend <-
+    pip(country = country, povline = povline, lkup = lkup)
+  res_pov_trend$pop_in_poverty <-
+    res_pov_trend$reporting_pop * res_pov_trend$headcount / pop_units
+  res_pov_trend <-
+    res_pov_trend[, c('country_code', 'reporting_year',
+                      'headcount', 'pop_in_poverty')]
+
+  # Fetch data for poverty bar chart
+  region <-
+    lkup$svy_lkup[country_code == country]$pcn_region_code %>%
+    unique()
+  countries <-
+    lkup$svy_lkup[pcn_region_code == region]$country_code %>%
+    unique()
+
+  res_pov_mrv <- pip(country = countries, lkup = lkup)
```

Comment:
Shouldn't this be:
`res_pov_mrv <- pip(country = countries, povline = povline, lkup = lkup)` Otherwise it will always use the `povline` default value and won't be sensitive to the poverty line chosen by the end-user.

Reply:
You are right. Good catch.